pkg/macaron/macaron.go

```diff
@@ -118,6 +118,19 @@ func FromContext(c context.Context) *Context {
 	return nil
 }
 
+type varsKey struct{}
+
+func Vars(r *http.Request) map[string]string {
```

Comment:
Nit. Is `Vars` a common name? What are you suggesting naming Query, or no changes needed there? 

Maybe some comments of these new methods would be good.

Reply:
I took it from the Gorilla/mux API. In fact the whole API of this PR is borrowed from Gorilla.

Reply:
Although you might be right: Gorilla uses `Vars`, while echo, gin-gonic and httprouter all use `Params`. I'll rename it.

Reply:
Okay, didn't know that :) It's not a big thing. But some comments of the functions would be good at least.